### PR TITLE
hello-rails-react

### PR DESCRIPTION
**Pull request by Abenezer-Tilahun**
### Description

- To build React components as a part of your Rails JavaScript and serve it in a component, react_component, to be used in a regular ERB file.

### General requirements

-  Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### Ruby requirements

- Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### Project requirements

- Create a new Rails app called 'hello-rails-react'.
- Initialize your project with Git.
-  Set up your project with web-packer the react-rails gems as describe in the previous lesson's tutorial.
- Make sure that your project has a Post-gres database set up.
- Create a table for storing your messages and create 5 different greetings.
- Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- Create a static view that will be the root of your app.
- Create the App component with react-router and render it in your static view.
- Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- Create a store, an action and a reducer that will connect to you API end-pont to get the random greeting.
- Display the random greeting in your Greeting component.

Thanks for reviewing the PR.

### CODE REVIEW

File: app/javascript/components/redux/greetings/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { applyMiddleware, combineReducers, legacy_createStore as createStore } from 'redux';
+import thunk from 'redux-thunk';
```
Comment (enkog) [inline app/javascript/components/redux/greetings/configureStore.js:2]: - [x] I recommend you run `npm i redux-thunk` to install this package as it is not installed on the application and so cannot be accessed here. To ascertain if a package is installed in your project, look through the `package.json`.

File: app/javascript/components/redux/greetings/configureStore.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import GreetingReducer from './greetings/greetings';
```
Comment (enkog) [inline app/javascript/components/redux/greetings/configureStore.js]: - [x] Please ensure to reference the correct path to the `GreetingsReducer`. 

**Current path in project to GreetingsReducer**
```import GreetingReducer from './greetings/greetings';```

**Correct path to GreetingsReducer**
```import GreetingReducer from './greetings';```

File: app/javascript/components/App.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from "react"
+import { Provider } from "react-redux";
+import {
+  BrowserRouter,
+  Routes,
+  Route,
+} from 'react-router-dom';
+import store from './redux/configureStore'
```
Comment (enkog) [inline app/javascript/components/App.js]: - [x] I suggest you also reference the correct path to the store as referencing an incorrect path will always result in errors.

**Current path in the project to store**
```import store from './redux/configureStore';```

**Correct path to store**
```import store from './redux/greetings/configureStore';```

File: README.md [hunk 4]
```diff
@@ -1 +1,41 @@
...
+
+- Ruby 3+
+- Rails 7
+- Postgresql
+- React
+
+## Setup
+
+- Clone the Repository
+- Run `bundle install` and `yarn install`
```
Comment (enkog) [inline README.md]: - [x] Please include instructions on how to create a greeting on the rails console. 
- I recommend you seed data in the database so there is a greeting displayed as the app is run for the first time.<br>

- [x] Please include detailed setup instructions to enable a visitor to your repository to set up and run your application without any hazzle.

**Hint:**
- Clone this repository with `https://github.com/Abenezer-Tilahun/Hello-world-rails-react.git` using your terminal or command line.
- Change to the project directory by entering `cd Hello-world-rails-react` in the terminal.
- Next run `bundle install` to install the necessary dependencies.
- run `rails db:create`
- Run `rails db:migrate` to setup your local database.
- Run `sudo service postgresql start` to start the database service.<br>
- Finally, run `rails s` to start the application.<br>

**Review by enkog — CHANGES_REQUESTED** [review]
Hi @Abenezer-Tilahun 👋 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [OPTIONAL] On installing the application I ran into the error below.

Capture

- However, I resolved it by running the following commands below.

  `yarn add @rails/webpacker`
  `bundle update webpacker`

- I suggest you include this in your README to guide visitors to your application in case they encounter the same error.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/components/redux/greetings/configureStore.js (hunk 1 above)
Comment (Abenezer-Tilahun) [reply app/javascript/components/redux/greetings/configureStore.js:2]: Hello @enkog  Thanks for suggestion I will fix it.

File: app/javascript/components/redux/greetings/configureStore.js (hunk 2 above)
Comment (Abenezer-Tilahun) [reply app/javascript/components/redux/greetings/configureStore.js]: Done

File: app/javascript/components/App.js (hunk 3 above)
Comment (Abenezer-Tilahun) [reply app/javascript/components/App.js]: Done

File: README.md (hunk 4 above)
Comment (Abenezer-Tilahun) [reply README.md]: Done

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi @Abenezer-Tilahun

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**